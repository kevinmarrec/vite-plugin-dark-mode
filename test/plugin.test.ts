import { createServer } from 'vite'
import { describe, expect, it } from 'vitest'

import plugin from '../src'

describe('plugin', () => {
  it('should inject dark mode script in index.html', async () => {
    const server = await createServer({ plugins: [plugin()] })
    const html = await server.transformIndexHtml('index.html', '<html><head></head><body></body></html>')

    expect(html).toMatchSnapshot()
  })
})
