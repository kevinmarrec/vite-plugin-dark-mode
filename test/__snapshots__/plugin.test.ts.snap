// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`plugin > should inject dark mode script in index.html 1`] = `
"<html><head>
  <script type="module" src="/@vite/client"></script>
  <script>(function () {
  const prefersDark = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches
  const setting = localStorage.getItem('vueuse-color-scheme') || 'auto'
  if (setting === 'dark' || (prefersDark && setting !== 'light'))
    document.documentElement.classList.toggle('dark', true)
})()</script>
</head><body></body></html>"
`;
